cart/cart.go

```diff
@@ -0,0 +1,81 @@
+package cart
+
+import (
+	"checkout_system/models"
+	"checkout_system/upload"
+	"github.com/sirupsen/logrus"
+)
+
+type ProductCart struct {
+	Products []models.Product `json:"products"`
+}
+
+var AvailableProducts upload.Products
+
+func New() *ProductCart {
+	err := AvailableProducts.LoadProductPrices()
+	if err != nil {
+		logrus.Error("an error occurred in LoadProductPrices():", err)
+		return nil
+	}
+	return new(ProductCart)
+}
+
+func (c *ProductCart) Contains(sku string) bool {
+	for _, p := range c.Products {
+		if p.SKU == sku {
+			return true
+		}
+	}
+	return false
+}
+
+func (c *ProductCart) Get(sku string) ProductCart {
+	var data []models.Product
+	for _, p := range c.Products {
+		if p.SKU == sku {
+			data = append(data, p)
+		}
+	}
+	return ProductCart{Products: data}
+}
+
+func (c ProductCart) Len() int {
+	return len(c.Products)
+}
+
+func (c *ProductCart) UpdateItemPrice(sku string, price float32) {
+	for i, p := range c.Products {
+		if p.SKU == sku {
+			c.Products[i].Price = price
+		}
+	}
+	return
+}
+
+func (c *ProductCart) UpdateOneItemPrice(sku string, price float32) {
+	for i, p := range c.Products {
+		if p.SKU == sku && p.Price != 0 {
+			c.Products[i].Price = price
+			return
+		}
+	}
+	return
+}
+
+func (c *ProductCart) Add(sku string) {
+	for _, p := range AvailableProducts.Products {
```

Comment:
Based on proposed change here https://github.com/ivankonevv/checkout_system/pull/1/files#diff-0d69c423b1cd4b04b1519c3239d7433d11783a85a2e13ad0eeb530e936bcc71fR10 we can simply search in map instead of loop through the list of products